### PR TITLE
feat: add a plugin bitbucket

### PR DESCRIPTION
**Pull request by thenicetgp**
## PreChecklist
firstly you should create a app_password for your account
https://bitbucket.org/account/settings/app-passwords/

#### bitbucket.go run configuration example
go run bitbucket.go -c 1 -o thenicetgp -r lake

## Summary

add bitbucket connection repo pr issue issue_comments pr_comments
add e2e tests: issue_test, pr_test, comment_test, account_test, repo_test, etc.

## Does this close any open issues?
#2100 

#### table "repos"
截屏2022-07-09 21 17 02

#### table "accounts"
截屏2022-08-16 09 47 06

#### table "issues"
截屏2022-08-16 09 44 56

#### table "pull_requests"
截屏2022-08-16 09 44 05

#### table "pull_request_comments"
截屏2022-08-16 09 45 52

#### table "issue_comments"
截屏2022-08-16 09 46 12

## Other information
Thanks to @mindlesscloud  for his guidance on my work.

### CODE REVIEW

**Comment by likyh** [comment]
Actually, you can convert a pr to a draft pr in the right pannel. 😁

**Comment by keon94** [comment]
Good work. Make sure you add e2e tests as well.

**Comment by thenicetgp** [comment]
> Good work. Make sure you add e2e tests as well.

thanks,I will add

File: plugins/bitbucket/api/blueprint.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		// handle taskOptions and transformationRules, by dumping them to taskOptions
+		transformationRules := make(map[string]interface{})
+		if len(scopeElem.Transformation) > 0 {
+			err = json.Unmarshal(scopeElem.Transformation, &transformationRules)
+			if err != nil {
+				return nil, err
+			}
+		}
+		// refdiff
+		if refdiffRules, ok := transformationRules["refdiff"]; ok {
```
Comment (keon94) [inline plugins/bitbucket/api/blueprint.go:51]: I'd make "refdiff" a global constant and put it in some shared package. The Github and Gitlab plugins also use it

Comment (keon94) [reply plugins/bitbucket/api/blueprint.go:51]: more generally, much of this transformation logic seems to be the same as those plugins. Would be good if you can refactor it into a common function

**Comment by keon94** [comment]
Hi @thenicetgp, could you give an overall status update on this PR? Are there more changes you are working on?

**Comment by thenicetgp** [comment]
> Hi @thenicetgp, could you give an overall status update on this PR? Are there more changes you are working on?

Hi @keon94 , thank for your comment,  this PR just add a plugin bitbucket, it can currently collect issues,pullrequests,accounts,repos,issue_comments and pr_comments. Domainlayer table accounts,repos,issue,pull_requests,issue_comments and pull_request_comments are filled. These tables have basically all the data they should have.
This PR makes no changes to the any other plugin.

File: plugins/bitbucket/api/blueprint.go (hunk 1 above)
Comment (thenicetgp) [reply plugins/bitbucket/api/blueprint.go:51]: got it , thanks for your remind.

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**